File: pkg/itypes/role.go
package itypes

type Role struct {
	ID                 int    `json:"id"`
	UUID               string `json:"uuid" sqlite:"text"`
	Position           int    `json:"position" sqlite:"int"`
	Name               string `json:"name" sqlite:"text"`
	Color              string `json:"color" sqlite:"text"`
	PermManageChannels uint8  `json:"perm_manage_channels" sqlite:"tinyint(1)"`
	PermManageRoles    uint8  `json:"perm_manage_roles" sqlite:"tinyint(1)"`
}

File: www/chat/js/api/role.js
"use strict";
//

//
/** @type {Map<string,Role} */
export const cache = new Map();

//
export class Role {
    constructor(o) {
        if (o === null) {
            this.is_null = true;
            return;
        }
        Object.assign(this, o);
        this.is_null = false;
        cache.set(this.uuid, this);
    }
}

//
new Role({
    uuid:"o",
    name:"Owner",
    color:"",
});
